File: exercicio e-commerce/php/panela.php
<?php
require_once 'Produto.php';

class Panela extends Produto {
    private $cor;
    private $modelo;

    public function __construct($cod, $cor, $modelo, $valor) {
        $this->setCod($cod);
        $this->cor = $cor;
        $this->modelo = $modelo;
        $this->setValor($valor);
    }

    public function cadastrar() {
        parent::cadastrar();
        $this->cor = readline('Digite a cor da panela: ');
        $this->modelo = readline('Digite o modelo da panela: ');
    }

    public function exibir() {
        echo $this->getCod() . " " . $this->cor . " " . $this->modelo . " " . $this->getValor() . "\n";
    }

    public function getCor() {
        return $this->cor;
    }

    public function setCor($cor) {
        $this->cor = $cor;
    }

    public function getModelo() {
        return $this->modelo;
    }

    public function setModelo($modelo) {
        $this->modelo = $modelo;
    }
}
?>

File: exercicio e-commerce/python/panela.py
from produto import Produto

class Panela(Produto):
    def __init__(self, cod, cor, modelo, valor):
        super().__init__()
        self.__cod = cod
        self.__cor = cor
        self.__modelo = modelo
        self.__valor = valor

    def cadastrar(self):
        super().cadastrar()
        self.__cor = input('Digite a cor da panela: ')
        self.__modelo = input('Digite o modelo da panela: ')

    def exibir(self):
        print(self.__cod, self.__cor, self.__modelo, self.__valor)

    @property
    def cor(self):
        return self.__cor

    @cor.setter
    def cor(self, cor):
        self.__cor = cor

    @property
    def modelo(self):
        return self.__modelo

    @modelo.setter
    def modelo(self, modelo):
        self.__modelo = modelo
